lib/bigtable/row.js

```diff
@@ -0,0 +1,618 @@
+var is = require('is');
+var extend = require('extend');
+var arrify = require('arrify');
+var flatten = require('lodash.flatten');
+
+/**
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * @private
+ */
+var Entry = require('./entry.js');
+
+/**
+ * @private
+ */
+var Filter = require('./filter.js');
+
+/**
+ * @type {module:common/grpcServiceObject}
+ * @private
+ */
+var GrpcServiceObject = require('../common/grpc-service-object.js');
+
+/**
+ * Create a Row object to interact with your table rows.
+ *
+ * @constructor
+ * @alias module:bigtable/row
+ *
+ * @example
+ * var gcloud = require('gcloud');
+ *
+ * var bigtable = gcloud.bigtable({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123',
+ *   cluster: 'zones/us-central1-b/clusters/gcloud-node'
+ * });
+ *
+ * var table = bigtable.table('presidents');
+ * var row = table.row('lincoln');
+ */
+function Row(table, name) {
+  var methods = {
+
+    /**
+     * Check if the row exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the row exists or not.
+     *
+     * @example
+     * row.exists(function(err, exists) {});
+     */
+    exists: true
+  };
+
+  var config = {
+    parent: table,
+    methods: methods,
+    id: name
+  };
+
+  GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Row, GrpcServiceObject);
+
+/**
+ * @private
+ */
+Row.formatChunks_ = function(chunks) {
+  chunks = chunks.reduce(function(chunkList, chunk) {
+    if (chunk.resetRow) {
+      return [];
+    }
+
+    if (chunk.rowContents) {
+      chunkList.push(chunk.rowContents);
+    }
+
+    return chunkList;
+  }, []);
+
+  var rowData = {};
+
+  chunks.forEach(function(chunk) {
+    var family = chunk.name;
+
+    rowData[family] = rowData[family] || {};
+
+    chunk.columns.forEach(function(column) {
+      var qualifier = Entry.convertFromBytes(column.qualifier);
+      column.cells.forEach(function(cell) {
+        var value = Entry.convertFromBytes(cell.value);
+        rowData[family][qualifier] = value;
+      });
+    });
+  });
+
+  return rowData;
+};
+
+/**
+ * Run a function to create a mutation batch request. Batching allows you to
+ * perform multiple operations in a single request. When you are finished making
+ * your changes, run the `done()` function provided in the callback function to
+ * complete the mutations. See an example below for more information.
+ *
+ * @param {function} fn - The function to create a batch request.
+ * @param {function} fn.done - Function to be called to finalize request.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * row.batch(function(done) {
```

Comment:
If our plan in the future to support promises is to return a promise when a callback is absent, that means with this proposed API change there is a conflict, since we can not return both a promise and some kind of batchable object.